tracing/tempo/http_client.go

```diff
@@ -0,0 +1,261 @@
+		return nil, errMarshal
+	}
+
+	return &response, nil
+}
+
+// convertBatchTrace Convert a trace returned by TraceQL query into a jaeger Trace
+func convertBatchTrace(trace otel.Trace, serviceName string) (jaegerModels.Trace, error) {
+
+	var jaegerModel jaegerModels.Trace
+
+	jaegerModel.TraceID = converter.ConvertId(trace.TraceID)
+	for _, span := range trace.SpanSet.Spans {
+		jaegerModel.Spans = append(jaegerModel.Spans, converter.ConvertSpanSet(span, serviceName, trace.TraceID, trace.RootTraceName)...)
+	}
+	jaegerModel.Matched = trace.SpanSet.Matched
+	jaegerModel.Processes = map[jaegerModels.ProcessID]jaegerModels.Process{}
+	jaegerModel.Warnings = []string{}
+
+	return jaegerModel, nil
+}
+
+// convertSingleTrace Convert a single trace returned by the TraceQL search endpoint
+func convertSingleTrace(traces *otelModels.Data, id string) (*model.TracingResponse, error) {
+	var response model.TracingResponse
+	var jaegerModel jaegerModels.Trace
+	serviceName := ""
+
+	jaegerModel.TraceID = converter.ConvertId(id)
+	if traces != nil {
+		serviceName = getServiceName(traces.Batches[0].Resource.Attributes)
+		for _, batch := range traces.Batches {
+			jaegerModel.Spans = append(jaegerModel.Spans, converter.ConvertSpans(batch.ScopeSpans[0].Spans, serviceName)...)
+		}
+		jaegerModel.Matched = len(jaegerModel.Spans)
+		jaegerModel.Processes = map[jaegerModels.ProcessID]jaegerModels.Process{}
+		jaegerModel.Warnings = []string{}
+
+	}
+
+	response.Data = append(response.Data, jaegerModel)
+
+	response.TracingServiceName = serviceName
+	return &response, nil
+}
+
+// prepareTraceQL returns a query in TraceQL format
+func prepareTraceQL(u *url.URL, tracingServiceName string, query models.TracingQuery) {
+	q := url.Values{}
+	q.Set("start", fmt.Sprintf("%d", query.Start.Unix()))
+	q.Set("end", fmt.Sprintf("%d", query.End.Unix()))
+	queryPart1 := TraceQL{operator1: ".service.name", operand: EQUAL, operator2: tracingServiceName}
+	queryPart2 := TraceQL{operator1: ".node_id", operand: REGEX, operator2: ".*"}
+	queryPart := TraceQL{operator1: queryPart1, operand: AND, operator2: queryPart2}
+
+	group1 := TraceQL{operator1: "status", operand: EQUAL, operator2: unquoted("error")}
+	group2 := TraceQL{operator1: "status", operand: EQUAL, operator2: unquoted("unset")}
+	group3 := TraceQL{operator1: "status", operand: EQUAL, operator2: unquoted("ok")}
+	groupQL := []TraceQL{group1, group2, group3}
+	group := Group{group: groupQL, operand: OR}
+
+	subquery := TraceQL{operator1: queryPart, operand: AND, operator2: group}
+	trace := TraceQL{operator1: Subquery{subquery}, operand: AND, operator2: Subquery{}}
+	queryQL := trace.getQuery()
+
+	q.Set("q", queryQL)
+	if query.MinDuration > 0 {
+		q.Set("minDuration", fmt.Sprintf("%dms", query.MinDuration.Milliseconds()))
+	}
+	if query.Limit > 0 {
+		q.Set("limit", strconv.Itoa(query.Limit))
+	}
+	u.RawQuery = q.Encode()
+	log.Debugf("Prepared Tempo API query: %v", u)
```

Comment:
How noisy will this be? Should be consider only logging this at Trace level?
```suggestion
	log.Tracef("Prepared Tempo API query: %v", u)
```

Reply:
I see that this is the same level as the Jaeger prepare query, https://github.com/josunect/kiali/blob/master/jaeger/http_client.go#L108
I thought it would make sense to have the same level in both?